django/contrib/postgres/constraints.py

```diff
@@ -178,3 +180,40 @@ def __repr__(self):
             '' if not self.include else ' include=%s' % repr(self.include),
             '' if not self.opclasses else ' opclasses=%s' % repr(self.opclasses),
         )
+
+    def validate(self, instance, exclude=None, using=DEFAULT_DB_ALIAS):
+        queryset = instance.__class__._default_manager.using(using)
+        replacement_map = {
+            field.name: Value(getattr(instance, field.attname), field)
+            for field in instance._meta.local_concrete_fields
+            if not exclude or field.name not in exclude
+        }
+        lookups = []
+        for idx, (expression, operator) in enumerate(self.expressions):
+            if isinstance(expression, str):
+                expression = F(expression)
+            if isinstance(expression, F):
+                if exclude and expression.name in exclude:
+                    return
+                rhs_expression = replacement_map.get(expression.name, expression)
+            else:
+                rhs_expression = expression.replace_references(replacement_map)
+                if exclude:
+                    for expr in rhs_expression.flatten():
+                        if isinstance(expr, F) and expr.name in exclude:
+                            return
+            lookup = PostgresOperatorLookup(lhs=expression, rhs=rhs_expression)
+            lookup.postgres_operator = operator
+            lookups.append(lookup)
+        queryset = queryset.filter(*lookups)
+        if not instance._state.adding and instance.pk is not None:
+            queryset = queryset.exclude(pk=instance.pk)
```

Comment:
That's a nice catch @kezabelle definitely something the logic should take into account that we should add tests for like 53fb45c6d8 did for ticket-17615.

The `validate` logic will need to know which model/table the constraint is _bound_ to in order to properly support mutli-table inheritance which it doesn't do at all right now.

For example, even putting the ticket-17615 issue aside, in the case of

```python
constraint = UniqueConstraint('x', 'y', name='unique_position')

class Position(models.Model):
    x = models.IntegerField()
    y = models.IntegerField()

    class Meta:
        contraints = [constraint]


class ImportantPosition(Positition):
    pass

Position.objects.create(x=1, y=2)
constraint.validate(ImportantPosition(x=1, y=2))
```

No validation error would be raised as we'd attempt

```sql
SELECT 1
FROM important_position
JOIN position ON (important_position.position_ptr_id = position_id)
WHERE x = 1 AND y 2
```

And since there's no existing `important_position` row the violation wouldn't be caught. What we need to perform here is

```sql
SELECT 1
FROM position
WHERE x = 1 AND y 2
``` 

Even when a child class instance is provided.

Ideally we'd bind `Constraint` classes to the model they are associated to at definition time (e.g. by setting `.model` on them like we do `Field` instances at `Option.contribute_to_class` time) so `.validate` can do `queryset = self.model._default_manager.using(using)` and `instance_pk = self._get_pk_val(self.model._meta)` but that will make testing of `Constraint` in isolation awkward (think of all the `Field.contribute_to_class` class in tests).

An alternative might be to change the `validate` signature to `(model: Type[Model], instance: Model, ...)` which is in line with `create_sql` and friends signature.

In all cases we need:
1. Addition regression test for perfoming validation on an instance of a model inheriting a constraint from a MTI parent
2. Adjustments of the validation logic to perform queries scoped to the model where the constraint was defined and not against the model of the provided instance.